docs/general/kakarot_components.md

```diff
@@ -29,43 +31,55 @@ contracts is stored as follows:
 For this latter computation to be account agnostic, Kakarot indeed uses a
 transparent proxy.
 
-## Contract Accounts
+## Accounts
 
-A _Contract Account_ is a StarkNet contract. However, it also acts as an
-Ethereum style contract account within the Kakarot EVM. In isolation it is not
-more than a StarkNet contract that stores some bytecode as well as some
-key-value pairs which were assigned to it on creation. It is only addressable
-via its StarkNet address and not an EVM address (which it is associated with
-inside the Kakarot EVM).
+The Ethereum protocol defines two kind of accounts:
 
-## Externally Owned Account
+- Externally Owned Accounts (EOA): accounts managed by a private-key and a
+  wallet
+- Contract Accounts (CA): accounts managed by their stored code (smart
+  contracts)
 
-Each Externally Owned Account in the EVM world has its counterpart in Starknet
-by the mean of a specific account contract deployed by Kakarot.
+Each of these account has a given Starknet contract class counterpart in
+Kakarot.
 
-This contract is a regular account contract in the Starknet sense with
-`__validate__` and `__execute__` entrypoint. However, it does decode and
-validate an EVM signed transaction and redirect it only to Kakarot. Further
-development will allow the user to have one single Starknet account for both
-Starknet native and Kakarot deployed dApp. For a general introduction to EVM
-transactions, see
-[the official doc](https://ethereum.org/en/developers/docs/transactions/).
+In the EVM, each account (EOA or CA) is located at an address and has the
+following properties:
+
+```json
+{
+  "balance": "0x0ba1a9ce0ba1a9ce",
+  "code": "0x6064640fffffffff2060005500",
+  "nonce": "0x00",
+  "storage": {
+    "0x01": "0x02"
+  }
+}
+```
 
-## Blockhash Registry
+Currently, Kakarot uses an external ERC0 contract for storing the balances and
```

Comment:
Proposal, something like:
> Currently, native coin balances (equivalent of ether) are stored as ERC20 balances, in a ERC20 token contract. Note users can still interact with native coin APIs on Kakarot (e.g. `payable address.transfer`). Consequently, only nonce, storage and code are stored in the corresponding Starknet contracts.

Reply:
not sure to follow how this is easier. The truth is that the ERC20 is really used as another storage contract, and this design doesn't make any sense outside of kakarot within starknet